### PR TITLE
feat(ui): Add logic to decide project details default charts

### PR DESCRIPTION
**Pull request by matejminar**
Users might have access to health/performance, but they might not use it (they might not be sending transactions/sessions).

Instead of showing empty default charts, we try to set the most sensible defaults for the optimal experience given the circumstances.

### CODE REVIEW

File: src/sentry/static/sentry/app/views/projectDetail/projectDetail.tsx [hunk 1]
```diff
@@ -131,12 +168,13 @@ class ProjectDetail extends AsyncView<Props, State> {
       selection,
     } = this.props;
     const project = this.project;
-    const hasPerformance = organization.features.includes('performance-view');
+    const {hasSessions} = this.state;
+    const hasTransactions = project?.firstTransactionEvent;
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/projectDetail/projectDetail.tsx:172]:
```suggestion
    const hasTransactions = !!project?.firstTransactionEvent;
```

File: src/sentry/static/sentry/app/views/projectDetail/projectDetail.tsx [hunk 2]
```diff
@@ -216,17 +254,23 @@ class ProjectDetail extends AsyncView<Props, State> {
...
                         location={location}
                         organization={organization}
                         router={router}
                         key={`project-charts-${id}`}
-                        index={id}
+                        chartId={id}
+                        chartIndex={index}
                         projectId={project?.id}
+                        hasSessions={hasSessions}
+                        hasTransactions={!!hasTransactions}
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/projectDetail/projectDetail.tsx:272]:
```suggestion
                        hasTransactions={hasTransactions}
```

**Review by priscilawebdev — APPROVED** [review]
lgtm

File: src/sentry/static/sentry/app/views/projectDetail/projectDetail.tsx (hunk 1 above)
Comment (matejminar) [reply src/sentry/static/sentry/app/views/projectDetail/projectDetail.tsx:172]: Later in the code, we care if it's boolean or undefined.

**Pull request merged**